apollo-router/src/plugins/telemetry/config_new/subgraph/selectors.rs

```diff
@@ -644,6 +651,56 @@ impl Selector for SubgraphSelector {
                     }
                 }
             }
+            SubgraphSelector::ResponseCacheStatus {
+                response_cache_status,
+                entity_type,
+            } => {
+                let cache_info: response_cache::plugin::CacheSubgraph = response
+                    .context
+                    .get(response_cache::metrics::CacheMetricContextKey::new(
+                        response.subgraph_name.clone(),
+                    ))
+                    .ok()
+                    .flatten()?;
+
+                let (cache_hit, cache_miss, entity_type_exist) = cache_info.0.iter().fold(
+                    (0, 0, false),
+                    |(mut cache_hit, mut cache_miss, mut entity_type_exist),
+                     (current_entity_type, cache_hit_miss)| {
+                        let compute = match entity_type {
+                            Some(EntityType::All(All::All)) | None => true,
+                            Some(EntityType::Named(entity_type_name)) => {
+                                current_entity_type == entity_type_name
+                            }
+                        };
+                        if compute {
+                            cache_hit += cache_hit_miss.hit;
+                            cache_miss += cache_hit_miss.miss;
+                            entity_type_exist = true;
+                        }
+
+                        (cache_hit, cache_miss, entity_type_exist)
+                    },
+                );
+                entity_type_exist.then(|| match response_cache_status {
+                    CacheStatus::Hit => (cache_hit > 0 && cache_miss == 0).into(),
+                    CacheStatus::Miss => (cache_hit == 0).into(),
```

Comment:
Should this be changed to match the `hit` and `partial_hit` behavior?
```suggestion
                    CacheStatus::Miss => (cache_hit == 0 && cache_miss > 0).into(),
```